src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -374,6 +374,13 @@ internal BoundStatement BindPossibleEmbeddedStatement(StatementSyntax node, Diag
         private BoundExpression BindThrownExpression(ExpressionSyntax exprSyntax, DiagnosticBag diagnostics, ref bool hasErrors)
         {
             var boundExpr = BindValue(exprSyntax, diagnostics, BindValueKind.RValue);
+            if (boundExpr.IsTypelessNew())
+            {
+                // NOTE This only disallows direct usage. One can simply bypass this by wrapping new() in a switch expression
```

Comment:
> One can simply bypass this by wrapping new() in a switch expression

Given that, is there any real reason to disallow? @jcouv

Reply:
Indeed, we should allow it. Confirmed with Neal. Since C# 8, the spec should say that there is a conversion to `System.Exception`, which would give the expression a target-type.